### PR TITLE
CBG-655: TestAllDocsOnly reliant on sequence numbers

### PR DESCRIPTION
**Pull request by sarathkumarsivan**

### CODE REVIEW

File: db/database_test.go [hunk 1]
```diff
@@ -674,28 +674,28 @@ func TestAllDocsOnly(t *testing.T) {
 	assert.Equal(t, 100, len(changes))
 
 	for i, change := range changes {
-		seq := i + 1
+		seq := i
```
Comment (adamcfraser) [inline db/database_test.go]: Should change this to something like 'docIndex' so that it's clear that it's now unrelated to sequence.

File: db/database_test.go [hunk 2]
```diff
@@ -674,28 +674,28 @@ func TestAllDocsOnly(t *testing.T) {
...
 		if i >= 23 {
 			seq++
 		}
-		assert.Equal(t, SequenceID{Seq: uint64(seq)}, change.Seq)
+		assert.Equal(t, i != 99, fmt.Sprintf("alldoc-%02d", seq) == change.ID)
```
Comment (adamcfraser) [inline db/database_test.go]: The way this section is written (both in the original test and now) is hard to follow.  I think it would be better written as:
```suggestion
                 if i == len(changes) - 1 {
                    // The last entry in the changes response should be the deleted document
                    assert.True(t, change.Deleted)
                    assert.Equal(t, "alldoc-23", change.ID)
                    assert.Equal(t, channels.SetOf(t, "all"), change.Removed)
                 } else {
                    // Verify correct ordering for all other documents
                    assert.Equal(t, fmt.Sprintf("alldoc-%02d", docIndex), change.ID)
                 }
```

File: db/database_test.go [hunk 3]
```diff
@@ -674,28 +674,28 @@ func TestAllDocsOnly(t *testing.T) {
...
 		assert.Equal(t, i == 99, change.Deleted)
 		var removed base.Set
 		if i == 99 {
 			removed = channels.SetOf(t, "all")
+			assert.Equal(t, "alldoc-23", change.ID)
 		}
-		goassert.DeepEquals(t, change.Removed, removed)
+		assert.Equal(t, removed, change.Removed)
```
Comment (adamcfraser) [inline db/database_test.go]: As discussed, it would also be useful to verify during this iteration that sequences are ascending for all entries in the changes response.

File: db/database_test.go [hunk 4]
```diff
@@ -674,28 +674,28 @@ func TestAllDocsOnly(t *testing.T) {
...
-	goassert.Equals(t, len(changes), 10)
+	assert.Len(t, changes, 10)
 	for i, change := range changes {
-		goassert.Equals(t, change.Seq, SequenceID{Seq: uint64(10*i + 1)})
-		goassert.Equals(t, change.ID, ids[10*i].DocID)
-		goassert.Equals(t, change.Deleted, false)
-		goassert.DeepEquals(t, change.Removed, base.Set(nil))
+		assert.Equal(t, ids[10*i].DocID, change.ID)
+		assert.False(t, change.Deleted)
+		assert.Equal(t, base.Set(nil), change.Removed)
```
Comment (adamcfraser) [inline db/database_test.go:707]: Could also validate ascending sequences here.

**Review by adamcfraser — CHANGES_REQUESTED** [review]
Generally looks good - a few suggestions.

File: db/database_test.go (hunk 1 above)
Comment (sarathkumarsivan) [reply db/database_test.go]: Sure Adam, this is committed to latest revision.

File: db/database_test.go (hunk 2 above)
Comment (sarathkumarsivan) [reply db/database_test.go]: Suggested change has been committed to latest revision.

File: db/database_test.go (hunk 3 above)
Comment (sarathkumarsivan) [reply db/database_test.go]: Added changes to check whether sequences are ascending for all entries in the changes response.

File: db/database_test.go (hunk 4 above)
Comment (sarathkumarsivan) [reply db/database_test.go:707]: Added changes to check whether sequences are ascending for all entries in the changes response.

**Review by sarathkumarsivan — COMMENTED** [review]
- Added changes to check whether sequences are ascending for all entries in the changes response.

**Review by adamcfraser — APPROVED** [review]
